libbeat/processors/add_locale/add_locale.go

```diff
@@ -89,7 +89,7 @@ func New(c *common.Config) (processors.Processor, error) {
 func (l addLocale) Run(event *beat.Event) (*beat.Event, error) {
 	zone, offset := time.Now().Zone()
 	format := l.Format(zone, offset)
-	event.PutValue("event.timezone", format)
+	event.Fields.Put("event.timezone", format)
```

Comment:
TBH I prefer that in general no one updates the` Fields` attribute directly. I especially would not put extra effort into changing processors that are not impacted by the bug.

Reply:
Well, if there is a strong opinion against it – I can rollback the change. But I still think it's way easier to navigate the code this way. For example, when you find usages of `PutValue` you find all the occurrences where it's meant to be used with metadata support, all the occurrences of `Fields.Put` indicate no metadata support.

After reading all the 39 processors and looking for this distinction I find this useful :)

Reply:
I prefer to have only one way of extending events with new fields. `PutValue` is safe for both fields and metadata. Please roll back these changes.

> After reading all the 39 processors and looking for this distinction I find this useful :)

This way other developers do not have to read all of the 39 processors to find this pattern. :) We should simplify processor development so there are fewer possibilities for error. Maybe it is also worth adding a few sentences to our developer guide (if we have any about processor development).